### PR TITLE
Fix edge agent connection (#3172)

### PR DESCRIPTION
**Pull request by ancaantochi**
* Recreate connection after every exception

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs [hunk 1]
```diff
@@ -264,7 +264,7 @@ async Task<Twin> GetTwinFunc()
             {
                 Events.ErrorGettingTwin(e);
 
-                if (!retrying && moduleClient != null && !(e is TimeoutException))
+                if (!retrying && moduleClient != null)
```
Comment (varunpuranik) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs:267]: unit tests?

Comment (ancaantochi) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs:267]: just added a test for TimeoutException

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/sdkClient/WrappingSdkModuleClient.cs [hunk 2]
```diff
@@ -14,7 +15,18 @@ public class WrappingSdkModuleClient : ISdkModuleClient
...
-        public Task OpenAsync() => this.sdkModuleClient.OpenAsync();
+        public Task OpenAsync()
+        {
+            try
+            {
+                return this.sdkModuleClient.OpenAsync();
+            }
+            catch (Exception)
+            {
+                this.sdkModuleClient?.Dispose();
```
Comment (philipktlin) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/sdkClient/WrappingSdkModuleClient.cs:26]: do we need to close first before calling dispose?

Comment (ancaantochi) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/sdkClient/WrappingSdkModuleClient.cs:26]: no need to close, confirmed with SDK team that we should use Dispose instead as is cleaning everything and Close has some gaps.

**Pull request merged**